pycyphal/transport/serial/_frame.py

```diff
@@ -132,58 +157,75 @@ def parse_from_cobs_image(image: memoryview) -> typing.Optional[SerialFrame]:
         return SerialFrame.parse_from_unescaped_image(memoryview(unescaped_image))
 
     @staticmethod
-    def parse_from_unescaped_image(header_payload_image: memoryview) -> typing.Optional[SerialFrame]:
+    def parse_from_unescaped_image(image: memoryview) -> typing.Optional[SerialFrame]:
         """
         :returns: Frame or None if the image is invalid.
         """
-        if len(header_payload_image) < _HEADER_SIZE:
-            return None
-
-        header = header_payload_image[:_HEADER_SIZE]
-        if not pycyphal.transport.commons.crc.CRC32C.new(header).check_residue():
-            return None
-
-        payload = header_payload_image[_HEADER_SIZE:]
-
-        # noinspection PyTypeChecker
-        (
-            version,
-            int_priority,
-            src_nid,
-            dst_nid,
-            int_data_spec,
-            transfer_id,
-            index_eot,
-        ) = _HEADER_WITHOUT_CRC_FORMAT.unpack_from(header)
-        if version != _VERSION:
+        try:
+            (
+                version,
+                int_priority,
+                source_node_id,
+                destination_node_id,
+                data_specifier_snm,
+                transfer_id,
+                frame_index_eot,
+                user_data,
+                header_crc,
+            ) = _HEADER_FORMAT.unpack_from(image)
+        except struct.error:
             return None
 
-        src_nid = None if src_nid == _ANONYMOUS_NODE_ID else src_nid
-        dst_nid = None if dst_nid == _ANONYMOUS_NODE_ID else dst_nid
-
-        try:  # https://github.com/OpenCyphal/pycyphal/issues/176
-            data_specifier: pycyphal.transport.DataSpecifier
-            if int_data_spec & (1 << 15) == 0:
-                data_specifier = pycyphal.transport.MessageDataSpecifier(int_data_spec)
-            else:
-                if int_data_spec & (1 << 14):
-                    role = pycyphal.transport.ServiceDataSpecifier.Role.RESPONSE
+        try:
+            if version == SerialFrame.VERSION:
+                crc = pycyphal.transport.commons.crc.CRC16CCITT()
+                crc.add(image[: _HEADER_FORMAT_NO_CRC.size])
+                if header_crc != crc.value:
+                    print("header CRC mismatch")
```

Comment:
![image](https://user-images.githubusercontent.com/3298404/226308336-d9b3c000-34bc-470e-a27d-c0fcad26b9b0.png)

Reply:
![image](https://user-images.githubusercontent.com/39975120/226324298-9b522340-9304-46ef-8f7f-6ba027eaaf79.png)

Reply:
I'll see what i can do

Reply:
I rather meant why is there a `print()` call...

Reply:
ah oops